bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingTypeResource.java

```diff
@@ -211,19 +206,26 @@ private ThingTypeDTO convertToThingTypeDTO(ThingType thingType, Locale locale) {
         return channelDefinitionDTOs;
     }
 
-    private Set<ThingTypeDTO> convertToThingTypeDTOs(List<ThingType> thingTypes, Locale locale) {
-        Set<ThingTypeDTO> thingTypeDTOs = new HashSet<>();
+    private Set<StrippedThingTypeDTO> convertToStrippedThingTypeDTOs(List<ThingType> thingTypes, Locale locale) {
+        Set<StrippedThingTypeDTO> strippedThingTypeDTOs = new HashSet<>();
 
         for (ThingType thingType : thingTypes) {
-            final ThingTypeDTO thingTypeDTO = convertToThingTypeDTO(thingType, locale);
-            if (thingTypeDTO != null) {
-                thingTypeDTOs.add(thingTypeDTO);
+            final StrippedThingTypeDTO strippedThingTypeDTO = convertToStrippedThingTypeDTO(thingType, locale);
+            if (strippedThingTypeDTO != null) {
+                strippedThingTypeDTOs.add(strippedThingTypeDTO);
             } else {
                 logger.warn("Cannot create DTO for thingType '{}'. Skip it.", thingType);
             }
         }
 
-        return thingTypeDTOs;
+        return strippedThingTypeDTOs;
     }
 
+    /*
+     * Convert thingType to StrippedDTO, which excludes configDescription, channels and properties
+     */
+    private StrippedThingTypeDTO convertToStrippedThingTypeDTO(ThingType thingType, Locale locale) {
```

Comment:
> Are we ok that the stripped version is REST specific and thus resides in io.rest and not in core.thing.dto?

This means that all "EnrichedDTO" and "DTOMappers" should be moved to core.thing.dto together with "official" DTO's?
> Additionally, the "Full" DTO cannot extend the strippedDTO

This crossed my mind. I was first thinking for the ThingTypeDTO to extend the StrippedDTO and the Enriched to extend the "Full" DTO... But not the best design.